libs/labelbox/src/labelbox/schema/workflow/workflow.py

```diff
@@ -0,0 +1,1504 @@
+        """
+        query_str = """
+        query GetProjectWorkflowPyApi($projectId: ID!) {
+          projectWorkflow(projectId: $projectId) {
+            projectId
+            config
+            createdAt
+            updatedAt
+          }
+        }
+        """
+
+        response = client.execute(query_str, {"projectId": project_id})
+        workflow_data = response["projectWorkflow"]
+        if workflow_data is None:
+            raise ValueError(f"Workflow not found for project ID: {project_id}")
+
+        # Ensure timezone info for proper parsing
+        created_at_str = workflow_data["createdAt"]
+        if not created_at_str.endswith("Z"):
+            created_at_str += "Z"
+        updated_at_str = workflow_data["updatedAt"]
+        if not updated_at_str.endswith("Z"):
+            updated_at_str += "Z"
+
+        return cls(
+            client=client,
+            project_id=workflow_data["projectId"],
+            config=workflow_data["config"],
+            created_at=datetime.fromisoformat(
+                created_at_str.replace("Z", "+00:00")
+            ),
+            updated_at=datetime.fromisoformat(
+                updated_at_str.replace("Z", "+00:00")
+            ),
+        )
+
+    def __init__(self, **data):
+        super().__init__(**data)
+
+        # Ensure config has required properties
+        if "config" not in data or data["config"] is None:
+            self.config = {"nodes": [], "edges": []}
+        elif not isinstance(self.config, dict):
+            self.config = {"nodes": [], "edges": []}
+        else:
+            # Ensure config.nodes exists
+            if "nodes" not in self.config:
+                self.config["nodes"] = []
+            # Ensure config.edges exists
+            if "edges" not in self.config:
+                logger.debug("Adding empty edges array to config")
+                self.config["edges"] = []
+
+        # Initialize edge factory
+        from labelbox.schema.workflow.edges import WorkflowEdgeFactory
+
+        self._edge_factory = WorkflowEdgeFactory(self)
+
+    def __repr__(self) -> str:
+        """Return a concise string representation of the workflow."""
+        num_nodes = len(self.config.get("nodes", []))
+        num_edges = len(self.config.get("edges", []))
+        return f"<ProjectWorkflow projectID:{self.project_id} nodes:{num_nodes} edges:{num_edges}>"
+
+    def __str__(self) -> str:
+        """Return a concise string representation of the workflow."""
+        num_nodes = len(self.config.get("nodes", []))
+        num_edges = len(self.config.get("edges", []))
+        return f"<ProjectWorkflow projectID:{self.project_id} nodes:{num_nodes} edges:{num_edges}>"
+
+    # Factory methods
+    @property
+    def create_edge(self):
+        """Factory method to create workflow edges between nodes."""
+        return self._edge_factory
+
+    @property
+    def WorkflowEdge(self):
```

Comment:
I prefer `create_edge`, also it's a bit unorthodox to have a property return a callable.

Reply:
Code refactored